indico/util/notification_templates.py

```diff
@@ -0,0 +1,113 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2016 European Organization for Nuclear Research (CERN).
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 3 of the
+# License, or (at your option) any later version.
+#
+# Indico is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Indico; if not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import unicode_literals
+
+from sqlalchemy.dialects.postgresql import ARRAY, JSON
+from sqlalchemy.ext.declarative import declared_attr
+
+from indico.core.db import db
+
+
+class NotificationTemplateBase(object):
+    """Base class for notification templates."""
+
+    #: The ID of the template
+    @declared_attr
```

Comment:
I had regular column definitions, but when I created the revision it game the error `sqlalchemy.exc.InvalidRequestError: Columns with foreign keys to other columns must be declared as @declared_attr callables on declarative mixin classes.`